crates/adapters/src/format/raw.rs

```diff
@@ -0,0 +1,531 @@
+                    endpoint_name,
+                    &e,
+                    request.query_string(),
+                )
+            })?,
+        ))
+    }
+
+    fn new_parser(
+        &self,
+        endpoint_name: &str,
+        input_stream: &InputCollectionHandle,
+        config: &YamlValue,
+    ) -> Result<Box<dyn Parser>, ControllerError> {
+        let config = RawParserConfig::deserialize(config).map_err(|e| {
+            ControllerError::parser_config_parse_error(
+                endpoint_name,
+                &e,
+                &serde_yaml::to_string(config).unwrap_or_default(),
+            )
+        })?;
+
+        let num_columns = input_stream.schema.fields.len();
+
+        if num_columns != 1 {
+            return Err(ControllerError::invalid_parser_configuration(
+                endpoint_name,
+                &format!("'raw' input format can only be used with tables that have a single column of type 'VARCHAR' or 'VARBINARY', but table '{}' has {num_columns} columns", input_stream.schema.name),
+            ));
+        }
+
+        let typ = input_stream.schema.fields[0].columntype.typ;
+
+        if !typ.is_varchar() && !typ.is_varbinary() {
+            return Err(ControllerError::invalid_parser_configuration(
+                endpoint_name,
+                &format!("'raw' input format can only be used with tables that have a single column of type 'VARCHAR' or 'VARBINARY', but table '{}' has a column of type {typ}", input_stream.schema.name),
+            ));
+        }
+
+        let input_stream = input_stream
+            .handle
+            .configure_deserializer(RecordFormat::Raw)?;
+        Ok(Box::new(RawParser::new(input_stream, config)) as Box<dyn Parser>)
+    }
+}
+
+struct RawParser {
+    /// Input handle to push parsed data to.
+    input_stream: Box<dyn DeCollectionStream>,
+
+    /// Whether the input starts with a header row.
+    config: RawParserConfig,
+
+    last_event_number: u64,
+}
+
+impl RawParser {
+    fn new(input_stream: Box<dyn DeCollectionStream>, config: RawParserConfig) -> Self {
+        Self {
+            input_stream,
+            last_event_number: 0,
+            config,
+        }
+    }
+
+    fn parse_record(&mut self, record: &[u8], errors: &mut Vec<ParseError>) {
+        if let Err(e) = self.input_stream.insert(record) {
+            errors.push(ParseError::bin_event_error(
+                format!("failed to deserialize raw record: {e}"),
+                self.last_event_number + 1,
+                record,
+                None,
+            ));
+        }
+    }
+}
+
+impl Parser for RawParser {
+    fn fork(&self) -> Box<dyn Parser> {
+        Box::new(Self::new(self.input_stream.fork(), self.config.clone()))
+    }
+
+    fn splitter(&self) -> Box<dyn super::Splitter> {
+        match self.config.mode {
+            RawParserMode::Lines => Box::new(LineSplitter),
+            RawParserMode::Blob => Box::new(Sponge),
```

Comment:
sponge? Is this a reference to Sponge Blob?

Reply:
Is absorbs everything :)